app/lib/widgets/add_farm.dart

```diff
@@ -45,15 +64,35 @@ class _NewFarmState extends State<NewFarm> {
     );
   }
 
+  Future<bool> isV4FarmAvailable(String name) async {
+    try {
+      final farms = await registrarClient!.farms
```

Comment:
what if the user didn't select a wallet? the registrarClient won't be defined yet

Reply:
The validation happens on click. That's why the wallet will always be there.